integrations.go

```diff
@@ -352,12 +354,12 @@ func (cfi *contextifyFramesIntegration) findNearbySourceCodeLocation(originalPat
 		possibleLocation := strings.Join(components, "/")
 
 		if fileExists(possibleLocation) {
-			cfi.cachedLocations[originalPath] = possibleLocation
+			cfi.updateCachedLocation(originalPath, possibleLocation)
```

Comment:
You need a `cfi.mu.RLock()` here.
If this collection isn't thread-safe, we must make sure no read happens while we're writing into it. If I understand correctly you are guarding the writes to it via `cfi.mu.Lock()` but not the reads with `cfi.mu.RLock()`.